src/main/java/com/jfrog/ide/idea/scan/ScannerBase.java

```diff
@@ -0,0 +1,376 @@
+    protected abstract AbstractInspection getInspectionTool();
+
+    protected void sendUsageReport() {
+        Utils.sendUsageReport(getPackageManagerName() + "-deps");
+    }
+
+    protected abstract String getPackageManagerName();
+
+    /**
+     * Groups a collection of DependencyNodes by the descriptor files of the modules that depend on them.
+     * The returned DependencyNodes inside the FileTreeNodes might be clones of the ones in depScanResults, but it's not
+     * guaranteed.
+     *
+     * @param depScanResults - collection of DependencyNodes.
+     * @param depMap         - a map of DependencyTree objects by their component ID.
+     * @return A list of FileTreeNodes (that are all DescriptorFileTreeNodes) having the DependencyNodes as their children.
+     */
+    protected abstract List<FileTreeNode> groupDependenciesToDescriptorNodes(Collection<DependencyNode> depScanResults, Map<String, List<DependencyTree>> depMap);
+
+    public abstract String getCodeBaseLanguage();
+
+    /**
+     * Scan and update dependency components.
+     *
+     * @param indicator - The progress indicator
+     */
+    private void scanAndUpdate(ProgressIndicator indicator) {
+        try {
+            // Building dependency tree
+            indicator.setText("1/3: Building dependency tree");
+            DependencyTree dependencyTree = buildTree();
+
+            // Sending the dependency tree to Xray for scanning
+            indicator.setText("2/3: Xray scanning project dependencies");
+            log.debug("Start scan for '" + basePath + "'.");
+            Map<String, DependencyNode> results = scanLogic.scanArtifacts(dependencyTree, serverConfig, indicator, prefix, this::checkCanceled);
+
+            indicator.setText("3/3: Finalizing");
+            if (results == null || results.isEmpty()) {
+                // No violations/vulnerabilities or no components to scan or an error was thrown
+                return;
+            }
+            Map<String, List<DependencyTree>> depMap = new HashMap<>();
+            mapDependencyTree(depMap, dependencyTree);
+
+            createImpactPaths(results, depMap, dependencyTree);
+            List<FileTreeNode> fileTreeNodes = new ArrayList<>(groupDependenciesToDescriptorNodes(results.values(), depMap));
+            addScanResults(fileTreeNodes);
+
+            // Source code scanning
+            List<FileTreeNode> sourceCodeResFileNodes = sourceCodeScannerManager.scanAndUpdate(indicator, results.values());
+            fileTreeNodes.addAll(sourceCodeResFileNodes);
+            addScanResults(fileTreeNodes);
+
+            // Force inspections run due to changes in the displayed tree
+            runInspections();
+
+        } catch (ProcessCanceledException e) {
+            log.info("Xray scan was canceled");
+        } catch (Exception e) {
+            logError(log, "Xray Scan failed", e, true);
+        } finally {
+            scanInProgress.set(false);
+            sendUsageReport();
```

Comment:
I know it is not your change, but I think that the usage report should be sent asynchronously at the beginning of the scan.